phoenix-ios/phoenix-ios/officers/PhotosManager.swift

```diff
@@ -0,0 +1,196 @@
+import SwiftUI
+
+fileprivate let filename = "PhotosManager"
+#if DEBUG && true
+fileprivate var log = LoggerFactory.shared.logger(filename, .trace)
+#else
+fileprivate var log = LoggerFactory.shared.logger(filename, .warning)
+#endif
+
+class PhotosManager {
+	
+	/// Singleton instance
+	public static let shared = PhotosManager()
+	
+	private let queue = DispatchQueue.init(label: "PhotosManager")
+	private let cache = Cache<String, UIImage>(countLimit: 30)
+	
+	private let memoryPressure: DispatchSourceMemoryPressure
+	
+	private init() { // must use shared instance
+		
+		memoryPressure = DispatchSource.makeMemoryPressureSource(
+			eventMask: [.warning, .critical],
+			queue: queue
+		)
+		memoryPressure.setEventHandler { [weak self] in
+			guard let self else {
+				return
+			}
+			let event = self.memoryPressure.data
+			switch event {
+				case .warning : self.respondToMemoryPressure(event)
+				case.critical : self.respondToMemoryPressure(event)
+				default       : break
+			}
+		}
+		memoryPressure.activate()
+	}
+	
+	// --------------------------------------------------
+	// MARK: Notifications
+	// --------------------------------------------------
+	
+	func respondToMemoryPressure(_ event: DispatchSource.MemoryPressureEvent) {
+		log.trace("respondToMemoryPressure()")
+		
+		// Note: This function is invoked on our `queue`, so we can safely modify the `cache` variable.
+		
+		cache.removeAll()
+	}
+	
+	// --------------------------------------------------
+	// MARK: Locations
+	// --------------------------------------------------
+	
+	lazy var photosDirectory: URL = {
```

Comment:
Is this `photosDirectory` accessible to other apps, or is it a private directory ?

Reply:
It's a private directory in the app's container